File: beta/paragraph_settings_align_left.js
function nicePara(inText, lineLength) {
    inText = inText.trim();
    var _ = [];
    var i;
    while (inText.length > 0) {
        if (inText.length <= lineLength) {
            _.push(inText);
            inText = "";
        } else {
            i = lineLength;
            if (inText[i] != " ") {
                while (!inText.slice(i - 3, i + 1).match(/[0-9a-zA-Z]{4}/) 
                && (inText[i - 1] != "-" && inText[i - 1] != " ")) {
                    i--;
                }
            }
            while (inText[i - 1] == " ") {
                i--;
            }
            if (inText[i] == " " | inText[i - 1] == "-") {
                _.push(inText.slice(0, i));
                inText = inText.slice(i).trim();
            } else {
                _.push(inText.slice(0, i - 1) + "-");
                inText = inText.slice(i - 1);
            }
        }
    }
    return _.join("\n");
}

File: beta/paragraph_settings_align_left.py
import re
def nicePara(inText, lineLength):
    inText = inText.strip()
    _ = []
    while len(inText) > 0:
        if len(inText) <= lineLength:
            _.append(inText)
            inText = ""
        else:
            i = lineLength
            if inText[i] != " ":
                while not re.match(r"[0-9a-zA-Z]{4}", inText[i - 3 : i + 1]) and inText[i - 1] != "-" and inText[i - 1] != " ":
                    i -= 1
            while inText[i - 1] == " ": i -= 1
            if inText[i] == " " or inText[i - 1] == "-":
                _.append(inText[0 : i])
                inText = inText[i:].strip()
            else:
                _.append(inText[0 : i - 1] + "-")
                inText = inText[i - 1:]
    return "\n".join(_)               